src/klein/resource.py

```diff
@@ -280,3 +280,39 @@ def write_response(r):
         d.addCallback(write_response).addErrback(log.err, _why="Unhandled Error writing response")
 
         return server.NOT_DONE_YET
+
+
+
+class KleinHTTPRequest(server.Request):
+
+    def getArg(self, key):
```

Comment:
I'm not a big fan of getters and setters either but it was the simple and easy to understand. I had a more elaborate solution using a dict-like object for the `args` variable but other devs weren't convinced with it and ultimatly it was scraped in favor of `getArgs`. As far as the Twisted mess, you provided a [ticket #228](https://tm.tl/288) but work on this might have stalled. Any hope of reviving that?